## Long read assembly
#### Assembly with `Canu`
Using long read sequencing data such as PacBio or nanopore in combination with high coverage illumina shot-gun sequencing for correction yields useful genome assemblies. For the first assembly, I used `Canu`, which allows to input sequencing data from different sources (PacBio, nanopore, BAC clones). 

```ShellSession
export HOME=/home/sk893857 #setting up home directory
canu -java=/tools/java/jre1.8.0_45/bin/java -p ./MySu_canu-v0 -d $HOME/genomes/Msulcatula genomeSize=40m gridEngineThreadsOption='--cpus-per-task=THREADS' maxMemory=120G maxThreads=8 -nanopore-raw 20171213_mysul-G10-S00.fastq.gz
```
#### Polish the assembly with `Pilon`
Then, the assembly should be polished using Illumina shot-gun sequencing data. First, the reads should be mapped to the new assembly and a sorted bam-file created with `samtools`. For polishing, I use `Pilon`.

1 - I trimmed the Illumina reads using `Trimmomatic0.36`.
```ShellSession
java -jar $HOME/bin/trimmomatic-0.36.jar PE -threads 8 -trimlog log-file 160316_SND405_A_L001_HZE-11_R1.fastq.gz 160316_SND405_A_L001_HZE-11_R2.fastq.gz HZE11_R1_trim.fq HZE11_R1_trim_uF.fq HZE11_R2_trim.fq HZE11_R2_trim_uR.fq ILLUMINACLIP:$HOME/trimmomatic-0.36/adapters/TruSeq2-PE.fa:5:30:10 SLIDINGWINDOW:3:18 LEADING:6 TRAILING:6 MINLEN:60
```
2 - I mapped the reads onto the draft assembly with `HISAT2`.
```ShellSession
hisat2-build MySu_canu-v0_contigs.fasta MySu_canu-v0_index
hisat2 -p 8 --no-spliced-alignment -k 10 -x MySu_canu-v0_index -1 HZE11_R1_trim.fq.gz -2 HZE11_R2_trim.fq.gz -S MySu_canu-v0.HZE11-map.sam
```
3 - Parsing the BAM file. 
```ShellSession
samtools sort -l 0 -o MySu_canu-v0.HZE11-map.bam -O bam MySu_canu-v0.HZE11-map.sam
samtools view -F 0x04 -b MySu_canu-v0.HZE11-map.bam > MySu_canu-v0.HZE11-map.aligned.bam
samtools index -b MySu_canu-v0.HZE11-map.aligned.bam
```
4 - I polished the assembly with `Pilon`.
```ShellSession
java -Xmx8G -jar $HOME/miniconda3/bin/pilon-1.22.jar --genome MySu_canu-v0_contigs.fasta --frags MySu_canu-v0.HZE11-map.aligned.bam --output canu_contigs.pilon.fa
```

In case of *Myriosclerotinia sulcatula*, I performed three rounds of polishing (including re-mapping of the Illumina reads to the polished versions), more rounds did not significantly improve the assembly any more. 
