## Illumina sequencing reads quality control

1 - I trimmed the Illumina reads using `Trimmomatic0.36`. This was done for all RNA-seq and DNA-seq raw files. 
```ShellSession
java -jar $HOME/bin/trimmomatic-0.36.jar PE -threads 8 -trimlog log-file 180307_SND393_A_L002_AOPH-7_R1.fastq.gz 180307_SND393_A_L002_AOPH-7_R2.fastq.gz AOPH-7_R1_trim.fq AOPH-7_R1_trim_uF.fq AOPH-7_R2_trim.fq AOPH-7_R2_trim_uR.fq ILLUMINACLIP:$HOME/trimmomatic-0.36/adapters/TruSeq2-PE.fa:5:30:10 SLIDINGWINDOW:3:18 LEADING:6 TRAILING:6 MINLEN:60
```

2 - Quality control via `fastQC`
```ShellSession
fastcq 180307_SND393_A_L002_AOPH-7_R1.fastq.gz 180307_SND393_A_L002_AOPH-7_R2.fastq.gz AOPH-7_R1_trim.fq.gz zAOPH-7_R2_trim.fq.gz -o fastqc_aoph07
```

2 - I mapped the reads onto the draft assembly with `HISAT2`.
```ShellSession
hisat2-build MySu_canu-v0_contigs.fasta MySu_canu-v0_index
hisat2 -p 8 --no-spliced-alignment -k 10 -x MySu_canu-v0_index -1 HZE11_R1_trim.fq.gz -2 HZE11_R2_trim.fq.gz -S MySu_canu-v0.HZE11-map.sam
```
3 - Parsing the BAM file. 
```ShellSession
samtools sort -l 0 -o MySu_canu-v0.HZE11-map.bam -O bam MySu_canu-v0.HZE11-map.sam
samtools view -F 0x04 -b MySu_canu-v0.HZE11-map.bam > MySu_canu-v0.HZE11-map.aligned.bam
samtools index -b MySu_canu-v0.HZE11-map.aligned.bam
```
