No assembly is perfect. To assess the quality of a genome assembly, several values can be assessed:

- Genome size (close to expectation?)
- number of contigs
- N50/N90 as measure of median contig size
- coverage
- completeness

#### Genome statistics
For a quick overview over some genome statistics, I used `assembly-stats`:
```ShellSession
assembly-stats Mysul_MySu01_v1.fa
```
The output includes N50, N90, and other size stats: 
```
sum = 43533135, n = 24, ave = 1813880.62, largest = 5732858
N50 = 2649720, n = 6
N60 = 2389439, n = 8
N70 = 2109168, n = 10
N80 = 1913538, n = 12
N90 = 1133061, n = 15
N100 = 13241, n = 24
N_count = 0
Gaps = 0
```

#### Re-mapping long reads
It can be useful to re-map long reads from PacBio or nanopore to assess the coverage of contigs. `bwa` is a tool that allows this:
```ShellSession
bwa index Mysul_MySu01_v1.fa
bwa mem -x ont2d Mysul_MySu01_v1.fa 20171213_mysul-G10-S00.fastq.gz > align-nanopore.sam
samtools sort -l 0 -o align-nanopore.bam -O bam align-nanopore.sam #create the BAM file
samtools view -F 0x04 -b align-nanopore.bam > align-nanopore.aligned.bam #clean up BAM file from non-mapping reads
```

#### BLASTN against NCBI nt database
Using `NCBI BLAST+ 2.7` and the latest version of the NCBI nucleotide collection database nt, I aligned the contigs against the known nucleotide sequences. This can help identify contigs of other origin (contaminant, viruses, mitochondrial genome). 
```ShellSession
blastn -task megablast -query Mysul_MySu01_v1.fa -db $HOME/blastdb/nt -outfmt '6 qseqid staxids bitscore std sscinames sskingdoms stitle' -culling_limit 5 -num_threads 32 -evalue 1e-25 -out MySu01_v1.blast-nt.tbl
```

#### Visualizing data with `Blobtools`
`Blobtools` is a very useful tool to summarize and visualize BLAST-nt and coverage data, and helps rapidly identify contaminant contigs. The putative identity of contigs can be generated on the levels of superkingdom, class, phylum, family, and genus.
```ShellSession
blobtools create -i Mysul_MySu01_v1.fa -b align-nanopore.aligned.bam -t MySu01_v1.blast-nt.tbl -o blobfile --db $HOME/miniconda3/opt/blobtools-1.0.1/data/nodesDB.txt #creates the JSON file
blobtools blobplot -i blobfile.blobDB.json -o MySu01_plot -r family --format pdf #to visualize as PDF
blobtools view -i blobfile.blobDB.json -r family -o MySu01_table #the table helps to inspect statistics (coverage, identity, GC content) of contigs
```

The plot looks like this:
![blobplot_mysul](https://github.com/stefankusch/Genome.assembly-Myriosclerotinia/blob/master/Mysul_MySu01_v1_blobtools_family.Mysul_MySu01_v1_blobtools.blobDB.json.bestsum.family.p7.span.100.blobplot.bam0.pdf)

This is the table:
```
## blobtools v1.0
## assembly	: /rwthfs/rz/cluster/home/sk893857/genomes/mysul/Mysul_MySu01_assembly-pilon.fasta
## coverage	: bam0 - /rwthfs/rz/cluster/home/sk893857/genomes/mysul/mapping/map.nanopore_mysul_v1.bam
## taxonomy	: tax0 - /rwthfs/rz/cluster/home/sk893857/genomes/mysul/blast/Mysul_MySu01_v1_blast-nt.tbl
## nodesDB	: /home/sk893857/miniconda3/opt/blobtools-1.0.1/data/nodesDB.txt
## taxrule	: bestsum
## min_score	: 0.0
## min_diff	: 0.0
## tax_collision_random	: False
##
# name	length	GC	N	bam0	family.t.6	family.s.7	family.c.8
tig00000001	5732858	0.4066	0	93.102	Sclerotiniaceae	5528231.0	2
tig00000002	2010949	0.4018	0	78.06	Sclerotiniaceae	2181427.0	2
tig00000003	4783387	0.4175	0	73.203	Sclerotiniaceae	5227003.0	5
tig00000007	684724	0.3724	0	74.655	Sclerotiniaceae	455470.0	2
tig00000008	3559358	0.4023	0	72.721	Sclerotiniaceae	3234598.0	5
tig00000014	1590599	0.4034	0	128.283	Sclerotiniaceae	2029212.0	17
tig00000017	2910650	0.3744	0	88.751	Sclerotiniaceae	2364855.0	14
tig00000021	2684408	0.3837	0	73.966	Sclerotiniaceae	2553810.0	2
tig00000024	2649720	0.3987	0	77.335	Sclerotiniaceae	2397076.0	5
tig00000027	2551437	0.4002	0	76.667	Sclerotiniaceae	2794938.0	4
tig00000029	2389439	0.4062	0	72.507	Sclerotiniaceae	2678392.0	8
tig00000033	313090	0.4088	0	80.036	Sclerotiniaceae	364039.0	0
tig00000042	1913538	0.3928	0	75.598	Sclerotiniaceae	1946320.0	13
tig00000059	1346483	0.4182	0	69.929	Sclerotiniaceae	1740451.0	12
tig00000066	1133061	0.3837	0	75.315	Sclerotiniaceae	1136511.0	11
tig00000068	1121634	0.3957	0	70.724	Sclerotiniaceae	1028544.0	29
tig00000075	932933	0.3825	0	77.475	Sclerotiniaceae	1270777.0	7
tig00000096	398532	0.4082	0	68.75	Sclerotiniaceae	466286.0	5
tig00000101	318643	0.4235	0	71.226	Sclerotiniaceae	296899.0	5
tig00000120	56788	0.2893	0	1012.365	Collemataceae	2736.0	1
tig00008973	2354695	0.3913	0	74.533	Sclerotiniaceae	2359012.0	6
tig00008974	15375	0.1582	0	32.007	no-hit	0.0	0
tig00008975	15213	0.2096	0	8.336	no-hit	0.0	0
tig00008976	2109168	0.3678	0	74.556	Sclerotiniaceae	1817311.0	17
tig00008977	13241	0.1605	0	19.068	no-hit	0.0	0
```
