#### Mapping RNA-seq data to the reference

First, I mapped available RNA-seq data to the reference genomes of *Sclerotinia sclerotiorum* and *S. trifoliorum* using `HISAT2`.
```ShellSession
hisat2-build Ssclerotiorum_v2_sorted.fasta Ss1980_index #for the S. sclerotiorum index
hisat2-build Sctrf_SwB9_assembly_v1.fa SwB9_index #for the S. trifoliorum index
#S. sclerotiorum Ss1980 mapping, camalexin data
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-1_R1_trim.fq.gz -2 AOPH-1_R2_trim.fq.gz -S AOPH-1.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-2_R1_trim.fq.gz -2 AOPH-2_R2_trim.fq.gz -S AOPH-2.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-3_R1_trim.fq.gz -2 AOPH-3_R2_trim.fq.gz -S AOPH-3.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-4_R1_trim.fq.gz -2 AOPH-4_R2_trim.fq.gz -S AOPH-4.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-5_R1_trim.fq.gz -2 AOPH-5_R2_trim.fq.gz -S AOPH-5.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-6_R1_trim.fq.gz -2 AOPH-6_R2_trim.fq.gz -S AOPH-6.sam
#S. trifoliorum SwB9 mapping, camalexin data
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-7_R1_trim.fq.gz -2 AOPH-7_R2_trim.fq.gz -S AOPH-7.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-8_R1_trim.fq.gz -2 AOPH-8_R2_trim.fq.gz -S AOPH-8.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-9_R1_trim.fq.gz -2 AOPH-9_R2_trim.fq.gz -S AOPH-9.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-10_R1_trim.fq.gz -2 AOPH-10_R2_trim.fq.gz -S AOPH-10.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-11_R1_trim.fq.gz -2 AOPH-11_R2_trim.fq.gz -S AOPH-11.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-12_R1_trim.fq.gz -2 AOPH-12_R2_trim.fq.gz -S AOPH-12.sam
#S. sclerotiorum Ss1980 mapping, plant infection data
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX13_trim.fq.gz -S HDX13.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX14_trim.fq.gz -S HDX14.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX15_trim.fq.gz -S HDX15.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX31_trim.fq.gz -S HDX31.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX32_trim.fq.gz -S HDX32.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX33_trim.fq.gz -S HDX33.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX34_trim.fq.gz -S HDX34.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX35_trim.fq.gz -S HDX35.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX36_trim.fq.gz -S HDX36.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX38_trim.fq.gz -S HDX38.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX39_trim.fq.gz -S HDX39.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX41_trim.fq.gz -S HDX41.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX42_trim.fq.gz -S HDX42.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX44_trim.fq.gz -S HDX44.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX45_trim.fq.gz -S HDX45.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX47_trim.fq.gz -S HDX47.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX48_trim.fq.gz -S HDX48.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX50_trim.fq.gz -S HDX50.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX51_trim.fq.gz -S HDX51.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX53_trim.fq.gz -S HDX53.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX54_trim.fq.gz -S HDX54.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX55_trim.fq.gz -S HDX55.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX56_trim.fq.gz -S HDX56.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX57_trim.fq.gz -S HDX57.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX58_trim.fq.gz -S HDX58.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX59_trim.fq.gz -S HDX59.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX60_trim.fq.gz -S HDX60.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG16_R1_trim.fq.gz -2 HWG16_R2_trim.fq.gz -S HWG16.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG17_R1_trim.fq.gz -2 HWG17_R2_trim.fq.gz -S HWG17.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG18_R1_trim.fq.gz -2 HWG18_R2_trim.fq.gz -S HWG18.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG19_R1_trim.fq.gz -2 HWG19_R2_trim.fq.gz -S HWG19.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG20_R1_trim.fq.gz -2 HWG20_R2_trim.fq.gz -S HWG20.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG21_R1_trim.fq.gz -2 HWG21_R2_trim.fq.gz -S HWG21.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG25_R1_trim.fq.gz -2 HWG25_R2_trim.fq.gz -S HWG25.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG26_R1_trim.fq.gz -2 HWG26_R2_trim.fq.gz -S HWG26.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG27_R1_trim.fq.gz -2 HWG27_R2_trim.fq.gz -S HWG27.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG28_R1_trim.fq.gz -2 HWG28_R2_trim.fq.gz -S HWG28.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG29_R1_trim.fq.gz -2 HWG29_R2_trim.fq.gz -S HWG29.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG30_R1_trim.fq.gz -2 HWG30_R2_trim.fq.gz -S HWG30.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG34_R1_trim.fq.gz -2 HWG34_R2_trim.fq.gz -S HWG34.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG35_R1_trim.fq.gz -2 HWG35_R2_trim.fq.gz -S HWG35.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG36_R1_trim.fq.gz -2 HWG36_R2_trim.fq.gz -S HWG36.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG37_R1_trim.fq.gz -2 HWG37_R2_trim.fq.gz -S HWG37.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG38_R1_trim.fq.gz -2 HWG38_R2_trim.fq.gz -S HWG38.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG39_R1_trim.fq.gz -2 HWG39_R2_trim.fq.gz -S HWG39.sam
```
I parsed the file to bam format and removed non-aligned reads for space-saving.
```ShellSession
for file in $HOME/*.sam; do
base=${file%.sam}
samtools sort -l 0 -o $base.bam -O bam $file;
samtools view -F 0x04 -b $base.bam > $base.aligned.bam;
mv $base.aligned.bam $base.bam;
samtools index -b $base.bam;
done
```
Next, I determined read counts for all genes with `HTseq` for downstream analysis. I moved the files for *S. sclerotiorum* and *S. trifoliorum* into separate folders before running the loop.
```ShellSession
for file in $HOME/*.bam; do
base=${file%.sam}
htseq-count -q -s no -f bam $file Ssclerotiorum_v2_gene_models.gtf > $base.counts.tab;
done
```
Lastly, I used `cuffnorm` to generate FPKM tables for all genes. 
```ShellSession
cuffnorm -q -p 8 -o ./map.counts/ \\
-L PDA_FRT_1,PDA_FRT_2,PDA_FRT_3,PDA_NEC_1,PDA_NEC_2,PDA_NEC_3,ATH_FRT_1,ATH_FRT_2,ATH_FRT_3,ATH_NEC_1,ATH_NEC_2,ATH_NEC_3,SLY_FRT_1,SLY_FRT_2,SLY_FRT_3,SLY_NEC_1,SLY_NEC_2,SLY_NEC_3,HAN_FRT_1,HAN_FRT_2,HAN_FRT_3,HAN_NEC_1,HAN_NEC_2,HAN_NEC_3,BVU_FRT_1,BVU_FRT_2,BVU_FRT_3,BVU_NEC_1,BVU_NEC_2,BVU_NEC_3,RCO_FRT_1,RCO_FRT_2,RCO_FRT_3,RCO_NEC_1,RCO_NEC_2,RCO_NEC_3,PVU_FRT_1,PVU_FRT_2,PVU_FRT_3,PVU_NEC_1,PVU_NEC_2,PVU_NEC_3 \\
--compatible-hits-norm \\
--library-norm-method classic-fpkm \\
Ssclerotiorum_v2_gene_models.gtf \\
./HDX55.bam, ./HDX56.bam, ./HDX57.bam, ./HDX58.bam, ./HDX59.bam, ./HDX60.bam, ./HDX34.bam, ./HDX35.bam, ./HDX36.bam, ./HDX31.bam, ./HDX32.bam, ./HDX33.bam, ./HDX47.bam, ./HDX50.bam, ./HDX53.bam, ./HDX48.bam, ./HDX51.bam, ./HDX54.bam, ./HDX38.bam, ./HDX41.bam, ./HDX44.bam, ./HDX39.bam, ./HDX42.bam, ./HDX45.bam, ./HWG16.bam, ./HWG17.bam, ./HWG18.bam, ./HWG19.bam, ./HWG20.bam, ./HWG21.bam, ./HWG25.bam, ./HWG26.bam, ./HWG27.bam, ./HWG28.bam, ./HWG29.bam, ./HWG30.bam, ./HWG34.bam, ./HWG35.bam, ./HWG36.bam, ./HWG37.bam, ./HWG38.bam, ./HWG39.bam
```
