#### Mapping RNA-seq data to the reference

First, I mapped available RNA-seq data to the reference genomes of *Sclerotinia sclerotiorum* and *S. trifoliorum* using `HISAT2`.
```ShellSession
hisat2-build Ssclerotiorum_v2_sorted.fasta Ss1980_index #for the S. sclerotiorum index
hisat2-build Sctrf_SwB9_assembly_v1.fa SwB9_index #for the S. trifoliorum index
#S. sclerotiorum Ss1980 mapping, camalexin data
hisat2 -p 8 --max-intronlen 1000 -k 1 -x Ss1980_index -1 AOPH-1_R1_trim.fq.gz -2 AOPH-1_R2_trim.fq.gz -S AOPH-1.sam
hisat2 -p 8 --max-intronlen 1000 -k 1 -x Ss1980_index -1 AOPH-2_R1_trim.fq.gz -2 AOPH-2_R2_trim.fq.gz -S AOPH-2.sam
hisat2 -p 8 --max-intronlen 1000 -k 1 -x Ss1980_index -1 AOPH-3_R1_trim.fq.gz -2 AOPH-3_R2_trim.fq.gz -S AOPH-3.sam
hisat2 -p 8 --max-intronlen 1000 -k 1 -x Ss1980_index -1 AOPH-4_R1_trim.fq.gz -2 AOPH-4_R2_trim.fq.gz -S AOPH-4.sam
hisat2 -p 8 --max-intronlen 1000 -k 1 -x Ss1980_index -1 AOPH-5_R1_trim.fq.gz -2 AOPH-5_R2_trim.fq.gz -S AOPH-5.sam
hisat2 -p 8 --max-intronlen 1000 -k 1 -x Ss1980_index -1 AOPH-6_R1_trim.fq.gz -2 AOPH-6_R2_trim.fq.gz -S AOPH-6.sam
#S. trifoliorum SwB9 mapping, camalexin data
hisat2 -p 8 --max-intronlen 1000 -k 1 -x SwB9_index -1 AOPH-7_R1_trim.fq.gz -2 AOPH-7_R2_trim.fq.gz -S AOPH-7.sam
hisat2 -p 8 --max-intronlen 1000 -k 1 -x SwB9_index -1 AOPH-8_R1_trim.fq.gz -2 AOPH-8_R2_trim.fq.gz -S AOPH-8.sam
hisat2 -p 8 --max-intronlen 1000 -k 1 -x SwB9_index -1 AOPH-9_R1_trim.fq.gz -2 AOPH-9_R2_trim.fq.gz -S AOPH-9.sam
hisat2 -p 8 --max-intronlen 1000 -k 1 -x SwB9_index -1 AOPH-10_R1_trim.fq.gz -2 AOPH-10_R2_trim.fq.gz -S AOPH-10.sam
hisat2 -p 8 --max-intronlen 1000 -k 1 -x SwB9_index -1 AOPH-11_R1_trim.fq.gz -2 AOPH-11_R2_trim.fq.gz -S AOPH-11.sam
hisat2 -p 8 --max-intronlen 1000 -k 1 -x SwB9_index -1 AOPH-12_R1_trim.fq.gz -2 AOPH-12_R2_trim.fq.gz -S AOPH-12.sam
```
I parsed the file to bam format and removed non-aligned reads for space-saving.
```ShellSession
for file in $HOME/*.sam; do
base=${file%.sam}
samtools sort -l 0 -o $base.bam -O bam $file;
samtools view -F 0x04 -b $base.bam > $base.aligned.bam;
mv $base.aligned.bam $base.bam;
samtools index -b $base.bam;
done
```
Next, I determined read counts for all genes with `HTseq` for downstream analysis. 
```ShellSession

```
Lastly, I used `cuffnorm` to generate FPKM tables for all genes. 
```ShellSession

```
