#### Mapping RNA-seq data to the reference

First, I mapped available RNA-seq data to the reference genomes of *Sclerotinia sclerotiorum* and *S. trifoliorum* using `HISAT2`.
```ShellSession
hisat2-build Ssclerotiorum_v2_sorted.fasta Ss1980_index #for the S. sclerotiorum index
hisat2-build Sctrf_SwB9_assembly_v1.fa SwB9_index #for the S. trifoliorum index
#S. sclerotiorum Ss1980 mapping, camalexin data
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-1_R1_trim.fq.gz -2 AOPH-1_R2_trim.fq.gz -S AOPH-1.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-2_R1_trim.fq.gz -2 AOPH-2_R2_trim.fq.gz -S AOPH-2.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-3_R1_trim.fq.gz -2 AOPH-3_R2_trim.fq.gz -S AOPH-3.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-4_R1_trim.fq.gz -2 AOPH-4_R2_trim.fq.gz -S AOPH-4.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-5_R1_trim.fq.gz -2 AOPH-5_R2_trim.fq.gz -S AOPH-5.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 AOPH-6_R1_trim.fq.gz -2 AOPH-6_R2_trim.fq.gz -S AOPH-6.sam
#S. trifoliorum SwB9 mapping, camalexin data
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-7_R1_trim.fq.gz -2 AOPH-7_R2_trim.fq.gz -S AOPH-7.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-8_R1_trim.fq.gz -2 AOPH-8_R2_trim.fq.gz -S AOPH-8.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-9_R1_trim.fq.gz -2 AOPH-9_R2_trim.fq.gz -S AOPH-9.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-10_R1_trim.fq.gz -2 AOPH-10_R2_trim.fq.gz -S AOPH-10.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-11_R1_trim.fq.gz -2 AOPH-11_R2_trim.fq.gz -S AOPH-11.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x SwB9_index -1 AOPH-12_R1_trim.fq.gz -2 AOPH-12_R2_trim.fq.gz -S AOPH-12.sam
#S. sclerotiorum Ss1980 mapping, plant infection data
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -U HDX40_trim.fq.gz -S HDX40.sam

hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG16_R1_trim.fq.gz -2 HWG16_R2_trim.fq.gz -S HWG16.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG17_R1_trim.fq.gz -2 HWG17_R2_trim.fq.gz -S HWG17.sam
hisat2 -p 8 --max-intronlen 500 -k 1 -x Ss1980_index -1 HWG18_R1_trim.fq.gz -2 HWG18_R2_trim.fq.gz -S HWG18.sam
```
I parsed the file to bam format and removed non-aligned reads for space-saving.
```ShellSession
for file in $HOME/*.sam; do
base=${file%.sam}
samtools sort -l 0 -o $base.bam -O bam $file;
samtools view -F 0x04 -b $base.bam > $base.aligned.bam;
mv $base.aligned.bam $base.bam;
samtools index -b $base.bam;
done
```
Next, I determined read counts for all genes with `HTseq` for downstream analysis. I moved the files for *S. sclerotiorum* and *S. trifoliorum* into separate folders before running the loop.
```ShellSession
for file in $HOME/*.bam; do
base=${file%.sam}
htseq-count -q -s no -f bam $file Ssclerotiorum_v2_gene_models.gtf > $base.counts.tab;
done
```
Lastly, I used `cuffnorm` to generate FPKM tables for all genes. 
```ShellSession
cuffnorm -q -p 8 -o ./map.counts/ -L HDX13,HDX14,HDX15,HDX16,HDX17,HDX18,HDX19,HDX20,HDX21,HDX22,HDX23,HDX24,HDX25,HDX26,HDX27,HDX31,HDX32,HDX33,HDX34,HDX35,HDX36,HDX38,HDX39,HDX41,HDX42,HDX44,HDX45,HDX47,HDX48,HDX50,HDX51,HDX53,HDX54,HDX55,HDX56,HDX57,HDX58,HDX59,HDX60,HWG16,HWG17,HWG18,HWG19,HWG20,HWG21,HWG25,HWG26,HWG27,HWG28,HWG29,HWG30,HWG34,HWG35,HWG36,HWG37,HWG38,HWG39 --compatible-hits-norm --library-norm-method classic-fpkm Ssclerotiorum_v2_gene_models.gtf ./HDX13.bam, ./HDX14.bam, ./HDX15.bam, ./HDX16.bam, ./HDX17.bam, ./HDX18.bam, ./HDX19.bam, ./HDX20.bam, ./HDX21.bam, ./HDX22.bam, ./HDX23.bam, ./HDX24.bam, ./HDX25.bam, ./HDX26.bam, ./HDX27.bam, ./HDX31.bam, ./HDX32.bam, ./HDX33.bam, ./HDX34.bam, ./HDX35.bam, ./HDX36.bam, ./HDX38.bam, ./HDX39.bam, ./HDX41.bam, ./HDX42.bam, ./HDX44.bam, ./HDX45.bam, ./HDX47.bam, ./HDX48.bam, ./HDX50.bam, ./HDX51.bam, ./HDX53.bam, ./HDX54.bam, ./HDX55.bam, ./HDX56.bam, ./HDX57.bam, ./HDX58.bam, ./HDX59.bam, ./HDX60.bam, ./HWG16.bam, ./HWG17.bam, ./HWG18.bam, ./HWG19.bam, ./HWG20.bam, ./HWG21.bam, ./HWG25.bam, ./HWG26.bam, ./HWG27.bam, ./HWG28.bam, ./HWG29.bam, ./HWG30.bam, ./HWG34.bam, ./HWG35.bam, ./HWG36.bam, ./HWG37.bam, ./HWG38.bam, ./HWG39.bam
```
