#### Genome completeness 
A very useful measure of genome quality is to use `BUSCO v3`. This algorithm searches for core conserved gene sets of lineages to assess the completeness of a given genome or proteome.  

```ShellSession
python $HOME/miniconda3/bin/BUSCO.py -f --out MySu01_v1.busco --in Mysul_MySu01_v1.fa --lineage ~/database/BUSCO/ascomycota_odb9 -m genome
```

The summary output of `BUSCO` reveals the number of complete, fragmented, duplicated, and missing core genes of the selected lineage, here ascomycota_odb9.
```
C:97.6%[S:97.6%,D:0.0%],F:1.0%,M:1.4%,n:1315

	1284	Complete BUSCOs (C)
	1284	Complete and single-copy BUSCOs (S)
	0	Complete and duplicated BUSCOs (D)
	13	Fragmented BUSCOs (F)
	18	Missing BUSCOs (M)
	1315	Total BUSCO groups searched
```

#### Repeat content
`RepeatMasker` allows the identification of known transposable elements and simple repetitive DNA. It also masks repetitive sequences.
```ShellSession
RepeatMasker -pa 8 -species fungi -gff -dir $HOME/RepeatMasker Mysul_MySu01_v1.fa
```
Note that I ran `RepeatMasker` with RepBase-20170127 and using version 4.0.7, since I lack access to later RepBase versions. 

`RepeatMasker` produces a GFF file of repeat elements, a genome file where repetitive elements are masked, and an output table looking like this:
```
==================================================
file name: Mysul_MySu01_v1.fa       
sequences:            24
total length:   43533135 bp  (43533135 bp excl N/X-runs)
GC level:         39.77 %
bases masked:    1550013 bp ( 3.56 %)
==================================================
               number of      length   percentage
               elements*    occupied  of sequence
--------------------------------------------------
Retroelements          289        59957 bp    0.14 %
   SINEs:                0            0 bp    0.00 %
   Penelope              1           44 bp    0.00 %
   LINEs:               23         1906 bp    0.00 %
    CRE/SLACS            2          125 bp    0.00 %
     L2/CR1/Rex          0            0 bp    0.00 %
     R1/LOA/Jockey       0            0 bp    0.00 %
     R2/R4/NeSL          0            0 bp    0.00 %
     RTE/Bov-B           0            0 bp    0.00 %
     L1/CIN4             0            0 bp    0.00 %
   LTR elements:       266        58051 bp    0.13 %
     BEL/Pao             0            0 bp    0.00 %
     Ty1/Copia         101        25856 bp    0.06 %
     Gypsy/DIRS1       164        32129 bp    0.07 %
       Retroviral        0            0 bp    0.00 %

DNA transposons         81        13198 bp    0.03 %
   hobo-Activator        6          342 bp    0.00 %
   Tc1-IS630-Pogo       29         9066 bp    0.02 %
   En-Spm                0            0 bp    0.00 %
   MuDR-IS905            0            0 bp    0.00 %
   PiggyBac              0            0 bp    0.00 %
   Tourist/Harbinger    14         1252 bp    0.00 %
   Other (Mirage,        0            0 bp    0.00 %
    P-element, Transib)

Rolling-circles          0            0 bp    0.00 %

Unclassified:            7          418 bp    0.00 %

Total interspersed repeats:       73573 bp    0.17 %


Small RNA:              30         4639 bp    0.01 %

Satellites:             46         5531 bp    0.01 %
Simple repeats:      25219      1276709 bp    2.93 %
Low complexity:       3238       190209 bp    0.44 %
==================================================

* most repeats fragmented by insertions or deletions
  have been counted as one element
                                                      

The query species was assumed to be fungi         
RepeatMasker Combined Database: Dfam_Consensus-20170127, RepBase-20170127
        
run with rmblastn version 2.6.0+
```
