This document is meant to summarize what input material I used and how I parsed the files to upload to an instance of `WebApollo`. 

## Input data
#### *Ab initio* gene predictions
`BRAKER` and `Augustus` produce gff annotation files of the predicted gene models. I have used the GeneMark and Augustus output produced by `BRAKER`, which needs parsing to gff3 as follows: 
```ShellSession
gffread -E MySu01_BRAKER-Augustus.gtf -o- > MySu01_BRAKER-Augustus.gff3
gffread -E MySu01_BRAKER-genemark.gtf -o- > MySu01_BRAKER-genemark.gff3
```
The gff3 file should look like this: 
```
# gffread -E genemark.gtf -o-
# gffread v0.10.1
##gff-version 3
tig00000001	GeneMark.hmm	mRNA	21	865	.	+	.	ID=1_t;geneID=1_g
tig00000001	GeneMark.hmm	exon	21	50	.	+	.	Parent=1_t
tig00000001	GeneMark.hmm	exon	147	865	.	+	.	Parent=1_t
tig00000001	GeneMark.hmm	CDS	21	50	.	+	2	Parent=1_t
tig00000001	GeneMark.hmm	CDS	147	865	.	+	2	Parent=1_t
```

#### Repeat landscape
The standard gff output file from `RepeatMasker` is already readable for `WebApollo`, but a little parsing is advised to make names of elements readable too. 
```ShellSession
sed 's/Target "/Name=/g' Mysul_MySu01_v1.fa.out.gff | sed 's/"//g' > MySu01_v1.RepeatMasker.gff
```
This gives a gff table looking like this:
```
tig00000001	RepeatMasker	similarity	1733	1779	15.5	+	.	Name=Motif:(CCATCC)n 1 48
tig00000001	RepeatMasker	similarity	1895	1932	19.1	+	.	Name=Motif:(AACAA)n 1 37
tig00000001	RepeatMasker	similarity	2058	2089	13.7	+	.	Name=Motif:(CCAAAT)n 1 32
```

#### BLAST similarities
Using BLAST to identify locations in the new genome with similarity to known genes can be a very useful hint, which is why I used published proteomes for a BLASTN against the new genome of *Myriosclerotinia sulcatula*, using the `NCBI BLAST+` package. 
```ShellSession
makeblastdb -in Mysul_MySu01_v1.fa -dbtype nucl -out MySu01_v1 -title "Msulcatula genome" # to create local database of Myriosclerotinia genome
blastn -evalue 1e-25 -query botrytis_cinerea_B05.10_transcripts.fasta -db $HOME/database/BLAST/MySu01_v1 -outfmt 6 -out Mysul_MySu01_v1_blastn_Bcinerea.tbl -num_threads 48
blastn -evalue 1e-25 -query Ssclerotiorum_1980_CDS.fa -db $HOME/database/BLAST/MySu01_v1 -outfmt 6 -out Mysul_MySu01_v1_blastn_Ss1980.tbl -num_threads 48
```
The BLAST results need to be parsed to gff so that `WebApollo` can visualize them. 
```ShellSession
cat Mysul_MySu01_v1_blastn_Bcinerea.tbl Mysul_MySu01_v1_blastn_Ss1980.tbl > Mysul_MySu01_v1_blastn_Sclerotiniaceae.tbl # combine BLAST results
blast2gff.py -b Mysul_MySu01_v1_blastn_Sclerotiniaceae.tbl > Mysul_MySu01_v1_blastn_Sclerotiniaceae.gff
```

#### RNA-seq reads
First, I mapped available RNA-seq data to the new genome Mysul_MySu01_v1.fa using `HISAT2`. RNA-seq raw data was trimmed via `Trimmomatic-0.36`. 
```ShellSession
hisat2-build Mysul_MySu01_v1.fa MySu01_index #for the index
hisat2 -p 8 --max-intronlen 500 -k 10 -x MySu01_index -1 AOPH-72_R1_trim.fq.gz -2 AOPH-72_R2_trim.fq.gz -S AOPH-72.sam
```
I parsed the file to bam format and to remove non-aligned reads.
```ShellSession
samtools sort -l 0 -o AOPH-72.bam -O bam AOPH-72.sam
samtools view -F 0x04 -b AOPH-72.bam > AOPH-72.aligned.bam
samtools index -b AOPH-72.aligned.bam
```

#### RNA-seq coverage
To visualize RNA-seq coverage in the `WebApollo` browser, I created a bigwig file using `bedtools2` and the UCSC script `bedGraphToBigWig`. 
```ShellSession
genomeCoverageBed -split -bg -ibam AOPH-72.aligned.bam -g Mysul_MySu01_v1.fa.fai | sort -k1,1 -k2,2n > AOPH-72.aligned.bedgraph
bedGraphToBigWig AOPH-72.aligned.bedgraph Mysul_MySu01_v1.fa.fai AOPH-72.aligned.bw
```

## Initializing genome in `WebApollo`
#### Setting up the genome
First, a directory for the new genome needs to be set-up with `mkdir ~/opt/Apollo-2.0.6/data_mysul`. Then, for uploading the genome:
```ShellSession
bin/prepare-refseqs.pl --fasta $HOME/data/Mysul_MySu01_v1.fa --out ~/opt/Apollo-2.0.6/data_mysul/
```

## Uploading data
#### Annotations (Repeats, gene predictions, BLAST)
In the Apollo directory, the annotations can be uploaded with the supplied scripts (`WebApollo-2.0.6`). The gene annotations were uploaded like this:
```ShellSession
bin/flatfile-to-json.pl -gff $HOME/data/MySu01_BRAKER-Augustus.gff3 --type mRNA -tracklabel BRAKER-Augustus --out ~/opt/Apollo-2.0.6/data_mysul/
bin/flatfile-to-json.pl -gff $HOME/data/MySu01_BRAKER-genemark.gff3 --type mRNA -tracklabel BRAKER-Genemark --out ~/opt/Apollo-2.0.6/data_mysul/
```
The Repeat landscape from `RepeatMasker` was uploaded as follows:
```ShellSession
bin/flatfile-to-json.pl -gff $HOME/data/MySu01_v1.RepeatMasker.gff --type similarity -tracklabel RepeatMasker --out ~/opt/Apollo-2.0.6/data_mysul/
```
BLAST results were added:
```ShellSession
bin/flatfile-to-json.pl -gff $HOME/data/MySu01.BLASTN.gff --type match_part -tracklabel BLASTN --out ~/opt/Apollo-2.0.6/data_mysul/
```
