After generating the manually-curated proteome, functional annotation is in order, including prediction of PFAM, secretion peptide, and transmembrane domains. 

#### PFAM
For PFAM annotations, I used `hmmscan` against the full database downloaded from the xfam page. To download the databank and create the `hmmscan` file:
```ShellSession
wget ftp://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/Pfam-A.hmm.gz #download
gunzip Pfam-A.hmm.gz #unzip the database
hmmpress Pfam-A.hmm #parse the database
```
And to run the scan against the query sequences:
```ShellSession
hmmscan --tblout MySu_v1.PFAM.txt Pfam-A.hmm MySu01_v1.proteins.fasta
```

#### Gene Ontology
I used **Blast2GO** for GO annotation, integrating InterProScan, BLAST and PFAM information to generate annotation files. 

#### Signal peptide
Signal peptides can be predicted using `SignalP4` (here v4.1). The script is simple:
```ShellSession
signalp -t euk -f short MySu01_v1.proteins.fasta > MySu01_v1.SigP41.out
```
`euk` defines eukaryotic organism. 

#### Transmembrane domains

One tool for this is `TMHMM`. 
```ShellSession
$HOME/bin/tmhmm MySu01_v1.proteins.fasta > MySu01_v1.TMHMM.out
```
The putative secretome should exclude proteins with putative transmembrane domains - I removed any protein with at least one TMhelix outside the secretion peptide to create the final list of secreted proteins of *Myriosclerotinia sulcatula*. 
